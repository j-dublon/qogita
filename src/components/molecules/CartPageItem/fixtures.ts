import { CartPageItemProps } from "@/types";

export const mockHandleUpdateCart = jest.fn();

export const mockProps: CartPageItemProps = {
  item: {
    name: "Mock Product",
    imageUrl: "http://your-product.com",
    recommendedRetailPrice: 52,
    price: 52,
    recommendedRetailPriceCurrency: "GBP",
    gtin: "987",
    id: "987",
    brandName: "Chanel",
    categoryName: "Perfume",
    quantity: 3,
  },
  handleUpdateCart: mockHandleUpdateCart,
};
