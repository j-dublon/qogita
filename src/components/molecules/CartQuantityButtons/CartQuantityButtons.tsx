import React, { FC } from "react";
import { CartQuantityButtonsProps } from "@/types";

const CartQuantityButtons: FC<CartQuantityButtonsProps> = ({
  numberInCart,
  updateCartQuantity,
}) => (
  <div className="flex flex-row gap-6 text-xl justify-self-center">
    <button
      disabled={numberInCart === 0}
      onClick={() => updateCartQuantity(numberInCart - 1)}
      className="text-2xl hover:text-blue-500"
    >
      -
    </button>
    <p>{numberInCart}</p>
    <button
      onClick={() => updateCartQuantity(numberInCart + 1)}
      className="text-2xl hover:text-blue-500"
    >
      +
    </button>
  </div>
);

export default CartQuantityButtons;
