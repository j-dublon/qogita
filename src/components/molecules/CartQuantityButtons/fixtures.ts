import { CartQuantityButtonsProps } from "@/types";

export const mockUpdateCartQuantity = jest.fn();

export const mockProps: CartQuantityButtonsProps = {
  numberInCart: 3,
  updateCartQuantity: mockUpdateCartQuantity,
};
