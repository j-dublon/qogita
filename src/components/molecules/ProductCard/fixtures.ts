import { Product } from "@/types";

export const mockProps: Product = {
  name: "Mock Product",
  imageUrl: "http://your-product.com",
  recommendedRetailPrice: 52,
  recommendedRetailPriceCurrency: "GBP",
  gtin: "987",
  brandName: "Chanel",
  categoryName: "Perfume",
};
