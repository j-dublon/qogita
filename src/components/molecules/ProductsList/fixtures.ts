import { ProductsListProps } from "@/types";

export const mockProps: ProductsListProps = {
  products: [
    {
      name: "Korres Endocare C Proteoglicanos Spf30 Ampoules 30x2ml",
      gtin: "8470001901682",
      recommendedRetailPrice: 59.99,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/cyW4Zm7C5EDgmWYLkGVQ6U.jpg",
      brandName: "Endocare",
      categoryName: "Women's Perfume",
    },
    {
      name: "Indasec Instituto Espa√±ol Argan Amphora Oil 400ml",
      gtin: "8411047133156",
      recommendedRetailPrice: 6.45,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/GPTKSirQGkJEKiS4PvAqtJ.jpg",
      brandName: "Indasec",
      categoryName: "Sundry",
    },
    {
      name: "Schwarzkopf 3d Mension Molding Wax - 100 Ml - Wax",
      gtin: "4045787263770",
      recommendedRetailPrice: 14.4,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/mv4rECdJqtdg2SnLM66LN8.jpg",
      brandName: "Schwarzkopf Professional",
      categoryName: "Hair Wax",
    },
    {
      name: "Veet Cold Wax Strips Armpits",
      gtin: "3059944026321",
      recommendedRetailPrice: 17.49,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/RoYyVChLGF94mZFtMTAHdi.jpg",
      brandName: "Veet",
      categoryName: "Depilatory Creams, Wax & Resin",
    },
    {
      name: "Philip B - Katira Hair Masque 178 Ml",
      gtin: "0893239000824",
      recommendedRetailPrice: 55.49,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/6dSNi2b8DsmaSRoNrDDNYX.jpg",
      brandName: "Philip B",
      categoryName: "Women's Perfume",
    },
    {
      name: "Electrolux Ekf7900 Counter Top Filter Coffee Maker 1.65 L Fully Automatic",
      gtin: "7332543331482",
      recommendedRetailPrice: 129.95,
      recommendedRetailPriceCurrency: "EUR",
      imageUrl:
        "https://images.qogita.com/files/images/variants/7eiCzQroBcf8NPLiqPWBN2.jpg",
      brandName: "Electrolux",
      categoryName: "Filter Coffee Makers",
    },
  ],
};
