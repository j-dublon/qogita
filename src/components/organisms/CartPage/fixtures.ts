import { CartPageProps } from "@/types";

export const mockProps: CartPageProps = {
  items: [
    {
      name: "Mock Product",
      imageUrl: "http://your-product.com",
      recommendedRetailPrice: 52,
      price: 52,
      recommendedRetailPriceCurrency: "EUR",
      gtin: "987",
      id: "987",
      brandName: "Chanel",
      categoryName: "Perfume",
      quantity: 3,
    },
    {
      name: "Mock Product Three",
      imageUrl: "http://your-product-three.com",
      recommendedRetailPrice: 12,
      price: 12,
      recommendedRetailPriceCurrency: "EUR",
      gtin: "123",
      id: "123",
      brandName: "Lynx",
      categoryName: "Deodorant",
      quantity: 2,
    },
  ],
  handleUpdateCart: jest.fn(),
  cartTotal: 356,
};
