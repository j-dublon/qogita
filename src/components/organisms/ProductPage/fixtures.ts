import { Product, ProductPageProps } from "@/types";
import { Item } from "react-use-cart";

export const mockAddItem = jest.fn();
export const mockUpdateItemQuantity = jest.fn();

export const mockProps: ProductPageProps = {
  product: {
    name: "Mock Product",
    imageUrl: "http://your-product.com",
    recommendedRetailPrice: 52,
    recommendedRetailPriceCurrency: "GBP",
    gtin: "987",
    brandName: "Chanel",
    categoryName: "Perfume",
  },
  numberInCart: 3,
  updateCartQuantity: jest.fn(),
};

export const mockCartItem = {
  ...mockProps.product,
  id: mockProps.product.gtin,
  price: mockProps.product.recommendedRetailPrice,
  quantity: 3,
};

export const alternativeProduct: Product = {
  name: "Other Mock Product",
  imageUrl: "http://your-product-two.com",
  recommendedRetailPrice: 35,
  recommendedRetailPriceCurrency: "GBP",
  gtin: "073",
  brandName: "Chanel",
  categoryName: "Perfume",
};

export const alternativeCartItem: Item = {
  name: "Other Mock Product",
  imageUrl: "http://your-product-two.com",
  recommendedRetailPrice: 35,
  price: 35,
  recommendedRetailPriceCurrency: "GBP",
  gtin: "073",
  id: "073",
  brandName: "Chanel",
  categoryName: "Perfume",
};
