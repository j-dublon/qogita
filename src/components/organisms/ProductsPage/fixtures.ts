import { Product, ProductsPageProps } from "@/types";

const mockProducts: Product[] = [
  {
    name: "Parodontax Duplo Herbal Fresh 75ml",
    gtin: "5054563079435",
    recommendedRetailPrice: 29.99,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/aB9r5isuPDUTTD3nLNsXvQ.jpg",
    brandName: "Parodontax",
    categoryName: "Toothpaste",
  },
  {
    name: "Poseidon The Black Men Edt Vapo 150 Ml - Beauty & Health",
    gtin: "8411047151242",
    recommendedRetailPrice: 22.99,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/co8e7Y9gf272e2W2LgA6fj.jpg",
    brandName: "Instituto Espanol",
    categoryName: "Men's Perfume",
  },
  {
    name: "Snowracer Classic - Sled - Black",
    gtin: "7313327300382",
    recommendedRetailPrice: 90.49,
    recommendedRetailPriceCurrency: "EUR",
    imageUrl:
      "https://images.qogita.com/files/images/variants/jLSEtd5DRU72VXf7ScZ2m9.jpg",
    brandName: "Stiga",
    categoryName: "Sledding",
  },
];

export const mockProps: ProductsPageProps = {
  loading: false,
  error: false,
  products: mockProducts,
  currentPage: 1,
  totalPages: 5,
  fetchData: jest.fn(),
};

export const mockPropsLoading: ProductsPageProps = {
  ...mockProps,
  loading: true,
};

export const mockPropsErrorState: ProductsPageProps = {
  ...mockProps,
  error: true,
};
