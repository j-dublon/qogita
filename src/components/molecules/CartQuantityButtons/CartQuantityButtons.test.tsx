import React from "react";
import { fireEvent, render } from "@testing-library/react";
import CartQuantityButtons from "./CartQuantityButtons";
import { mockProps, mockUpdateCartQuantity } from "./fixtures";

describe("Component: CartQuantityButtons", () => {
  it("SHOULD match snapshot", () => {
    const component = render(<CartQuantityButtons {...mockProps} />);
    expect(component).toMatchSnapshot();
  });

  it("SHOULD display number in cart correctly WHEN rendered", () => {
    const { getByText } = render(<CartQuantityButtons {...mockProps} />);
    expect(getByText(mockProps.numberInCart));
  });

  it("SHOULD call updateCartQuantity with correct values WHEN minus button is clicked", () => {
    const { getByText } = render(<CartQuantityButtons {...mockProps} />);

    fireEvent.click(getByText("-"));
    expect(mockUpdateCartQuantity).toHaveBeenCalledWith(
      mockProps.numberInCart - 1
    );
  });

  it("SHOULD call updateCartQuantity with correct values WHEN plus button is clicked", () => {
    const { getByText } = render(<CartQuantityButtons {...mockProps} />);

    fireEvent.click(getByText("+"));
    expect(mockUpdateCartQuantity).toHaveBeenCalledWith(
      mockProps.numberInCart + 1
    );
  });
});
