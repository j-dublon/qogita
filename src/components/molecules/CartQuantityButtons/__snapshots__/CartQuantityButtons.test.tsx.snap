// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CartQuantityButtons SHOULD match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-row gap-6 text-xl justify-self-center"
      >
        <button
          class="text-2xl hover:text-blue-500"
        >
          -
        </button>
        <p>
          3
        </p>
        <button
          class="text-2xl hover:text-blue-500"
        >
          +
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-row gap-6 text-xl justify-self-center"
    >
      <button
        class="text-2xl hover:text-blue-500"
      >
        -
      </button>
      <p>
        3
      </p>
      <button
        class="text-2xl hover:text-blue-500"
      >
        +
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
